api/lib/opentelemetry/trace/span_context.rb

```diff
@@ -6,24 +6,32 @@
 
 module OpenTelemetry
   module Trace
-    # A SpanContext contains the state that must propagate to child @see Spans and across process boundaries.
-    # It contains the identifiers (a @see TraceId and @see SpanId) associated with the @see Span and a set of
-    # @see TraceOptions.
+    # A SpanContext contains the state that must propagate to child {Span}s and across process boundaries.
+    # It contains the identifiers (a trace ID and span ID) associated with the {Span} and a set of
+    # {TraceFlags}.
     class SpanContext
-      attr_reader :trace_id, :span_id, :trace_options
+      attr_reader :trace_id, :span_id, :trace_flags
 
       def initialize(
-        trace_id: generate_trace_id,
-        span_id: generate_span_id,
-        trace_options: TraceOptions::DEFAULT
+        trace_id: OpenTelemetry::Trace.generate_trace_id,
+        span_id: OpenTelemetry::Trace.generate_span_id,
+        trace_flags: OpenTelemetry::Trace::TraceFlags::DEFAULT
       )
         @trace_id = trace_id
         @span_id = span_id
-        @trace_options = trace_options
+        @trace_flags = trace_flags
       end
 
       def valid?
-        !(@trace_id.zero? || @span_id.zero?)
+        @trace_id != OpenTelemetry::Trace::INVALID_TRACE_ID &&
+          @span_id != OpenTelemetry::Trace::INVALID_SPAN_ID
+      end
+
+      def self.invalid
```

Comment:
I'm not really happy about this. I'd prefer an `INVALID` or `DEFAULT` constant, but the `require` order makes this difficult (`new` tries to access `OpenTelemetry::Trace::TraceFlags::DEFAULT` which hasn't been defined yet).